### DISCUSSION TITLE
Cannot create directed graph

### DISCUSSION THREAD

**Discussion by tyler828** [FLAMEGPU/FLAMEGPU2#1178] [Q&A] [answered]
Hello, I am trying to define a directed graph for my agents to traverse through by following this doc page: [https://docs.flamegpu.com/guide/environment/index.html#defining-a-directed-graph](url)

> // Fetch the model's environment
flamegpu::EnvironmentDescription env = model.Environment();
// Declare a new directed graph named 'fgraph'
EnvironmentDirectedGraphDescription fgraph = model.Environment().newDirectedGraph("fgraph");

I am getting errors trying to define the graph, `error: identifier "EnvironmentDirectedGraphDescription" is undefined` and `error: class "flamegpu::EnvironmentDescription" has no member "newDirectedGraph"`

**Comment by Robadob** [comment]
Are you using a recent copy of FLAMEGPU2? That feature was only merged mid December (https://github.com/FLAMEGPU/FLAMEGPU2/pull/1089).

> `error: class "flamegpu::EnvironmentDescription" has no member "newDirectedGraph"` 

It definitely exists in the current state of `master` branch.

https://github.com/FLAMEGPU/FLAMEGPU2/blob/25a37160e68d077ba9cb1c683530ec27e92e8f83/include/flamegpu/model/EnvironmentDescription.h#L222

Try updating your copy of FLAMEGPU2 from git before rebuilding.

**Reply by tyler828** [reply, marked as answer]
I found the issue, I simply had to update the tag in my CMakeLists.txt.  The template project still has version v2.0.0-rc in its CMake file.

**Reply by Robadob** [reply]
Ah we will have forgotten to update that. I'll get it fixed right away.

Thanks
